app/scripts/modules/core/delivery/service/executions.transformer.service.js

```diff
@@ -164,7 +166,17 @@ module.exports = angular.module('spinnaker.core.delivery.executionTransformer.se
 
         var currentStage = lastRunningStage || lastFailedStage || lastNotStartedStage || lastStage;
         stage.status = currentStage.status;
-        stage.endTime = currentStage.endTime;
+        // if a stage is running, ignore the endTime of the parent stage
+        if (!currentStage.endTime) {
+          delete stage.endTime;
+        }
+        let lastEndingStage = _.max(stages, 'endTime');
+        // if the current stage has an end time (i.e. it failed or completed), use the maximum end time
+        // of all the child stages as the end time for the parent - we do this because the parent might
+        // have been an initialization stage, which ends within a few milliseconds
+        if (currentStage.endTime && lastEndingStage.endTime) {
```

Comment:
Is there a Jira to fix this in Orca?  Might be a good I idea to add a reference  to it the PR or as a TODO in the code so we have some traceability.